src/woo_publications/publications/admin.py

```diff
@@ -47,17 +48,25 @@ def sync_to_index(
 
     num_objects = filtered_qs.count()
 
-    if model is Publication:
-        task_fn = index_publication
-        kwarg_name = "publication_id"
-    elif model is Document:
-        task_fn = index_document
-        kwarg_name = "document_id"
-    else:  # pragma: no cover
-        raise ValueError("Unsupported model: %r", model)
+    if model not in [Publication, Document]:  # pragma: no cover
+        raise ValueError("Unknown model: %r", model)
 
     for obj in filtered_qs.iterator():
-        transaction.on_commit(partial(task_fn.delay, **{kwarg_name: obj.pk}))
+        if model is Publication:
+            transaction.on_commit(
+                partial(index_publication.delay, publication_id=obj.pk)
+            )
+        elif model is Document:
+            document_url = absolute_document_download_uri(
+                request, document_uuid=obj.uuid
+            )
+            transaction.on_commit(
+                partial(
+                    index_document.delay, document_id=obj.pk, download_url=document_url
+                )
+            )
+        else:  # pragma: no cover
+            assert False, "unreachable"
```

Comment:
Opted to use this patern instead of assert_never(model) because of the following error that it caused pyright to return 

` Argument of type "Unknown | type[Publication] | type[Document]" cannot be assigned to parameter "arg" of type "Never" in function "assert_never"
    Type "Unknown | type[Publication] | type[Document]" is not assignable to type "Never"`